lightning-background-processor/Cargo.toml

```diff
@@ -17,7 +17,14 @@ rustdoc-args = ["--cfg", "docsrs"]
 bitcoin = "0.29.0"
 lightning = { version = "0.0.111", path = "../lightning", features = ["std"] }
 lightning-rapid-gossip-sync = { version = "0.0.111", path = "../lightning-rapid-gossip-sync" }
-futures = { version = "0.3", optional = true }
+futures = { version = "=0.3.17", optional = true }
+futures-core = { version = "=0.3.17", optional = true }
+futures-task = { version = "=0.3.17", optional = true }
+futures-channel = { version = "=0.3.17", optional = true }
+futures-executor = { version = "=0.3.17", optional = true }
+futures-io = { version = "=0.3.17", optional = true }
+futures-sink = { version = "=0.3.17", optional = true }
+futures-util = { version = "=0.3.17", optional = true }
```

Comment:
Mh, I think it at least needs the `async-await-macro` feature since `select` is a re-export from `futures-macro`. I think I might have a version working with 1.41.1, but we'll have to wait until CI stops randomly failing to find out...

Reply:
Hm, not sure how you made it work locally without default features (namely `std` and  `async-await-macro`, which depends on `futures-macros`), since `select` is only re-exported when they are set, see:

https://github.com/rust-lang/futures-rs/blob/7a98cf0bbeb397dcfaf5f020b371ab9e836d33d4/futures-util/src/async_await/select_mod.rs#L315-L325

Reply:
Also note that unpinning, i.e., `version = "0.3"` (https://github.com/lightningdevkit/rust-lightning/pull/1744/commits/abb55fde9e9eff8d2bb90277b07eef8c810270a9), leads to version 0.3.24 being used, which fails on 1.41.1.

Reply:
Alright, now went this way with https://github.com/lightningdevkit/rust lightning/pull/1744/commits/19ae0b6d86e5cd3ef62a81c18b4e57cf9c46af20. 

Unfortunately, `default-features = false` doesn't work for `lightning-block-sync` though, since `Mutex` also requires `std`: https://github.com/rust-lang/futures-rs/blob/7a98cf0bbeb397dcfaf5f020b371ab9e836d33d4/futures-util/src/lock/mod.rs#L15-L18